recon.py

```diff
@@ -0,0 +1,329 @@
+"""
+A Proposal for reconciliation graph and reconciliation interface.
+"""
+from abc import ABC  # Abstract Base Classes
+from typing import List, Dict
+from enum import Enum
+
+__all__ = ['MappingNode', 'EventType', 'Event', 'Cospeciation', 'Duplication', 'Transfer', 'Loss',
+           'TipTip', 'Reconciliation', 'ReconGraph']
+
+class MappingNode:
+    """
+    MappingNode is a node in the reconciliation graph that maps
+    a parasite to a host.
+    """
+    def __init__(self, parasite_vertex: str, host_vertex: str):
+        self._parasite = parasite_vertex
+        self._host = host_vertex
+
+    @property
```

Comment:
`@property` allows us to call `mapping_node.host` instead of `mapping_node.host()`.